### PR TITLE
Disable transformation validation

### PR DESCRIPTION
**Pull request by EItanya**
# Description

Allow disabling of Transformation Validation. This feature currently uses an external envoy binary, which can be quite slow when called many times during a single sync. This setting will allow this feature to be optionally turned off.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/kube2e/gateway/gateway_test.go [hunk 1]
```diff
@@ -1460,6 +1460,95 @@ spec:
...
+				settingsClientFactory := &factory.KubeResourceClientFactory{
+					Crd:         gloov1.SettingsCrd,
+					Cfg:         cfg,
+					SharedCache: kube.NewKubeCache(ctx),
+				}
+
+				settingsClient, err = gloov1.NewSettingsClient(ctx, settingsClientFactory)
+				Expect(err).NotTo(HaveOccurred())
+				err = settingsClient.Register()
+				Expect(err).NotTo(HaveOccurred())
```
Comment (kdorosh) [inline test/kube2e/gateway/gateway_test.go:1483]: consider using `clienthelpers.MustSettingsClient(ctx)` instead

File: test/kube2e/gateway/gateway_test.go [hunk 2]
```diff
@@ -1460,6 +1460,95 @@ spec:
...
+				Expect(settingsList).To(HaveLen(1))
+				settings := settingsList[0]
+				settings.Gateway.Validation.DisableTransformationValidation = nil
+				_, err = settingsClient.Write(settings, clients.WriteOpts{
+					OverwriteExisting: true,
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				cancel()
```
Comment (kdorosh) [inline test/kube2e/gateway/gateway_test.go:1512]: we don't want to call cancel twice?

File: test/kube2e/gateway/gateway_test.go [hunk 3]
```diff
@@ -1460,6 +1460,95 @@ spec:
...
+			})
+
+			It("will not reject invalid transformation", func() {
+				injaTransform := `{% if default(data.error.message, "") != "" %}400{% else %}{{ header(":status") }}{% endif %`
```
Comment (kdorosh) [inline test/kube2e/gateway/gateway_test.go:1516]: lets add a comment that notes that the ending curly brace is missing, hence this is invalid

File: install/helm/gloo/generate/values.go [hunk 4]
```diff
@@ -237,12 +237,13 @@ type ServiceAccount struct {
...
-	FailurePolicy         string   `json:"failurePolicy" desc:"failurePolicy defines how unrecognized errors from the Gateway validation endpoint are handled - allowed values are 'Ignore' or 'Fail'. Defaults to Ignore "`
-	Webhook               *Webhook `json:"webhook" desc:"webhook specific configuration"`
+	Enabled                         bool     `json:"enabled" desc:"enable Gloo Edge API Gateway validation hook (default true)"`
+	AlwaysAcceptResources           bool     `json:"alwaysAcceptResources" desc:"unless this is set this to false in order to ensure validation webhook rejects invalid resources. by default, validation webhook will only log and report metrics for invalid resource admission without rejecting them outright."`
+	AllowWarnings                   bool     `json:"allowWarnings" desc:"set this to false in order to ensure validation webhook rejects resources that would have warning status or rejected status, rather than just rejected."`
+	DisableTransformationValidation bool     `json:"disableTransformationValidation" desc:"set this to true to "`
```
[the remark this answers is not in the document]
Comment (kdorosh) [reply install/helm/gloo/generate/values.go:243]: also add full description

File: projects/gloo/api/v1/settings.proto [hunk 5]
```diff
@@ -536,6 +536,13 @@ message GatewayOptions {
         //   - regex routes that make later routes unreachable
         //   - duplicate matchers
         google.protobuf.BoolValue warn_route_short_circuiting = 8;
+
+        // By default gloo will attempt to validate transformations by calling out to a local envoy binary in `validate` mode.
+        // Calling this local envoy binary can become slow when done many times during a single validation.
+        // Setting this to true will stop gloo from calling out to envoy to validate the transformations, which may speed up the
```
Comment (saiskee) [inline projects/gloo/api/v1/settings.proto:542]: speed up? Or skip transformation validation completely?

File: projects/gloo/pkg/plugins/transformation/plugin.go [hunk 6]
```diff
@@ -173,6 +176,14 @@ func (p *Plugin) convertTransformation(
 	return ret
 }
 
+func (p *Plugin) validateTransformation(ctx context.Context, transformations *envoytransformation.RouteTransformations) error {
+	err := bootstrap.ValidateBootstrap(ctx, p.settings, bootstrap.BuildPerFilterBootstrapYaml(FilterName, transformations))
```
Comment (saiskee) [inline projects/gloo/pkg/plugins/transformation/plugin.go:180]: Can we just check if `if settings.GetGateway().GetValidation().GetDisableTransformationValidation().GetValue() {` here? and use that to determine if we call `validateBootstrap` or not

Comment (saiskee) [reply projects/gloo/pkg/plugins/transformation/plugin.go:180]: So we're not running boostrap.BuildPerFilterBootstrapYaml even if we don't need it

Comment (EItanya) [reply projects/gloo/pkg/plugins/transformation/plugin.go:180]: Sure, I'm gonna refactor it a bit. I want the logic to be as close to the source as possible so anytime this code is called that check is used

File: projects/gloo/api/v1/settings.proto (hunk 5 above)
Comment (EItanya) [reply projects/gloo/api/v1/settings.proto:542]: Speed up general validation time by skipping transformation validation

**Review by saiskee — APPROVED** [review]

**Pull request merged**